spritecontrol.cpp

```diff
@@ -105,8 +104,13 @@ bool SpriteChoreographer::should_change_pattern() {
 }
 
 void SpriteChoreographer::change_pattern() {
-	size_t rand = (size_t) (Noise::random() * m_enabled_patterns.size());
-	m_pattern = m_enabled_patterns.at(rand);
+	std::vector<PatternName> patterns = { };
```

Comment:
A better name for this is probably `candidate_patterns` or similar, just calling it `patterns` is really undescriptive.